cmd/prometheus/main_test.go

```diff
@@ -238,6 +238,36 @@ func TestWALSegmentSizeBounds(t *testing.T) {
 	}
 }
 
+func TestMaxChunkSize(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+
+	for _, size := range []string{"1MB", "512MB", "1GB"} {
+		prom := exec.Command(promPath, "-test.main", "--storage.tsdb.max-chunk-size="+size, "--config.file="+promConfig)
```

Comment:
Does it make sense to really have unit test that just starts Prometheus with different value for this option? If this option is not set properly (does not change any segment size), this test would pass anyway, right? 


I would remove it and rather clone some more advanced tests like querying block/WAL with different segment size.

Reply:
+1, we dont need a test for the flags. Tests with different sizes could go in the tsdb package.

Reply:
Testing that different allocation sizes works is already widely done (that's why I didn't have to wite NewWriterWithSegSize myself), though fair enough that we don't actually test that the setting took.

Reply:
Alright, changed this test to verify the new lower bound requested by codesome. Testing configurable segment sizes is already done in TestChunkWriter_ReadAfterWrite so not much point in doing that again. Technically that leaves the plumbing from the DB to the ChunkWriter untested, but that doesn't seem all that concerning.

Reply:
Going to mark this conversation as resolved just so there's a clean slate and hopefully can get eyes on this again. If you still have issues with the current state feel free to poke me about it.